sdk/python/feast/infra/offline_stores/redshift.py

```diff
@@ -163,20 +240,192 @@ def to_s3(self) -> str:
             self._s3_path,
             self._config.offline_store.iam_role,
             self.query,
+            self._drop_columns,
         )
         return self._s3_path
 
     def to_redshift(self, table_name: str) -> None:
         """ Save dataset as a new Redshift table """
+        query = f'CREATE TABLE "{table_name}" AS ({self.query});\n'
+        if self._drop_columns is not None:
+            for column in self._drop_columns:
+                query += f"ALTER TABLE {table_name} DROP COLUMN {column};\n"
+
         aws_utils.execute_redshift_statement(
             self._redshift_client,
             self._config.offline_store.cluster_id,
             self._config.offline_store.database,
             self._config.offline_store.user,
-            f'CREATE TABLE "{table_name}" AS ({self.query})',
+            query,
         )
 
 
+# This is based on sdk/python/feast/infra/offline_stores/bigquery.py:MULTIPLE_FEATURE_VIEW_POINT_IN_TIME_JOIN
+# There are couple of changes from BigQuery:
+# 1. Use VARCHAR instead of STRING type
+# 2. Use DATEADD(...) instead of Timestamp_sub(...)
+# 3. Replace `SELECT * EXCEPT (...)` with `SELECT *`, because `EXCEPT` is not supported by Redshift.
+#    Instead, we drop the column later after creating the table out of the query.
+# We need to keep this query in sync with BigQuery.
```

Comment:
Please note the changes that landed with #1712

Reply:
I already rebased on that and made sure to update the Redshift query accordingly.